src/Akka.Persistence.Azure/Snapshot/AzureBlobSnapshotStore.cs

```diff
@@ -123,190 +125,127 @@ protected override void PreStart()
         protected override async Task<SelectedSnapshot> LoadAsync(string persistenceId,
             SnapshotSelectionCriteria criteria)
         {
-            var requestOptions = GenerateOptions();
-            BlobResultSegment results = null;
-            using (var cts = new CancellationTokenSource(_settings.RequestTimeout))
+            using var cts = new CancellationTokenSource(_settings.RequestTimeout);
             {
-                results = await Container.ListBlobsSegmentedAsync(SeqNoHelper.ToSnapshotSearchQuery(persistenceId),
-                    true,
-                    BlobListingDetails.Metadata, null, null, requestOptions, new OperationContext(), cts.Token);
-            }
+                var results = Container.GetBlobsAsync(
+                    prefix: SeqNoHelper.ToSnapshotSearchQuery(persistenceId), 
+                    traits: BlobTraits.Metadata,
+                    cancellationToken: cts.Token);
 
-            // if we made it down here, the initial request succeeded.
+                var pageEnumerator = results.AsPages().GetAsyncEnumerator(cts.Token);
 
-            async Task<SelectedSnapshot> FilterAndFetch(BlobResultSegment segment)
-            {
+                if (!await pageEnumerator.MoveNextAsync())
```

Comment:
We're only interested on the first page, will not retrieve the whole list of blobs